client/client.go

```diff
@@ -224,8 +231,10 @@ func (c *httpClusterClient) Do(ctx context.Context, act httpAction) (*http.Respo
 	var body []byte
 	var err error
 
-	for _, ep := range eps {
-		hc := c.clientFactory(ep)
+	perm := c.rand.Perm(leps)
```

Comment:
Do we really need to do this? 

1. Perm is not a common operation. 
2. I would expect switchToFirst takes the position argument. But it takes the actual value.

It seems like we only need to shuffle the endpoints when setting. And always starts with the pinned one. It is more clean and easy to read in my opinion.